vsphere/resource_vsphere_license_test.go

```diff
@@ -0,0 +1,225 @@
+package vsphere
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/license"
+)
+
+var (
+	testAccLabels = map[string]string{
+		"VpxClientLicenseLabel": "Hello World",
+		"TestTitle":             "FooBar",
+	}
+)
+
+func TestAccVSphereLicenseBasic(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccVSpherePreLicenseBasicCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccVSphereLicenseDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVSphereLicenseBasicConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccVSphereLicenseExists("vsphere_license.foo"),
+				),
+			},
+		},
+	})
+
+}
+
+func TestAccVSphereLicenseInvalid(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVSphereLicenseInvalidConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccVSphereLicenseNotExists("vsphere_license.foo"),
+				),
+				ExpectError: regexp.MustCompile("License file not found"),
+			},
+		},
+	})
+
+}
+
+func TestAccVSphereLicenseWithLabels(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccVSpherePreLicenseBasicCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVSphereLicenseWithLabelConfig(testAccLabels),
+				Check: resource.ComposeTestCheckFunc(
+					testAccVSphereLicenseWithLabelExists("vsphere_license.foo"),
+				),
+			},
+		},
+	})
+
+}
+
+func testAccVSphereLicenseInvalidConfig() string {
+
+	// quite sure this key cannot be valid
+	return `resource "vsphere_license" "foo" {
+  					license_key = "HN422-47193-58V7M-03086-0JAN2"
+			}`
+}
+
+func testAccVSphereLicenseWithLabelConfig(labels map[string]string) string {
+
+	// precheck already checks if this is present or not
+	key := os.Getenv("VSPHERE_LICENSE")
+
+	labelString := labelToString(labels)
+
+	return fmt.Sprintf(`resource "vsphere_license" "foo" {
+							license_key = "%s"
+
+							labels {
+								%s
+							}		 	
+						}`,
+		key, labelString)
+}
+
+func labelToString(labels map[string]string) string {
+	val := ""
+	for key, value := range labels {
+		val += fmt.Sprintf(`
+			%s = "%s"
+		`, key, value)
+
+	}
+	return val
+}
+
+func testAccVSphereLicenseBasicConfig() string {
```

Comment:
As per the label generator here, remember we should style this function the same way:

```go
func testAccVSphereLicenseWithLabelConfig() string {
  return fmt.Sprintf(`
resource "vsphere_license" "foo" {
  license_key = "%s"
}
`, os.Getenv("VSPHERE_LICENSE"))
}
```